net/freeradius/src/opnsense/mvc/app/views/OPNsense/Freeradius/dhcp.volt

```diff
@@ -0,0 +1,115 @@
+POSSIBILITY OF SUCH DAMAGE.
+
+#}
+
+<script>
+
+    $( document ).ready(function() {
+        /*************************************************************************************************************
+         * link grid actions
+         *************************************************************************************************************/
+
+        $("#grid-dhcps").UIBootgrid(
+            {   'search':'/api/freeradius/dhcp/searchDhcp',
+                'get':'/api/freeradius/dhcp/getDhcp/',
+                'set':'/api/freeradius/dhcp/setDhcp/',
+                'add':'/api/freeradius/dhcp/addDhcp/',
+                'del':'/api/freeradius/dhcp/delDhcp/',
+                'toggle':'/api/freeradius/dhcp/toggleDhcp/'
+            }
+        );
+
+        /*************************************************************************************************************
+         * Commands
+         *************************************************************************************************************/
+
+        /**
+         * Reconfigure
+         */
+        $("#reconfigureAct").click(function(){
+            $("#reconfigureAct_progress").addClass("fa fa-spinner fa-pulse");
+            ajaxCall(url="/api/freeradius/service/reconfigure", sendData={}, callback=function(data,status) {
+                // when done, disable progress animation.
+                $("#reconfigureAct_progress").removeClass("fa fa-spinner fa-pulse");
+
+                if (status != "success" || data['status'] != 'ok') {
+                    BootstrapDialog.show({
+                        type: BootstrapDialog.TYPE_WARNING,
+                        title: "{{ lang._('Error reconfiguring FreeRADIUS') }}",
+                        message: data['status'],
+                        draggable: true
+                    });
+                } else {
+                    ajaxCall(url="/api/freeradius/service/reconfigure", sendData={});
+                }
+            });
+        });
+
+    });
+
+
+</script>
+
+<ul class="nav nav-tabs" data-tabs="tabs" id="maintabs">
+    <li class="active"><a data-toggle="tab" href="#dhcps">{{ lang._('DHCP') }}</a></li>
+</ul>
+<div class="tab-content content-box tab-content">
+    <div id="dhcps" class="tab-pane fade in active">
+        <!-- tab page "dhcps" -->
+        <table id="grid-dhcps" class="table table-condensed table-hover table-striped table-responsive" data-editDialog="dialogEditFreeRADIUSDhcp">
+            <thead>
+            <tr>
+                <th data-column-id="enabled" data-type="string" data-formatter="rowtoggle">{{ lang._('Enabled') }}</th>
+                <th data-column-id="gatewayip" data-type="string" data-visible="true">{{ lang._('Gateway IP') }}</th>
+                <th data-column-id="dns1" data-type="string" data-visible="true">{{ lang._('DNS1') }}</th>
+                <th data-column-id="dns2" data-type="string" data-visible="true">{{ lang._('DNS2') }}</th>
```

Comment:
Does this primary/secondary have any configurational limitation? otherwise it could just be a CSV field and you could show it in one column. Primary server will be the first one. ;)

Reply:
ok, this looks better now :)